desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.java

```diff
@@ -1027,26 +1027,33 @@ private JsonElement getVotingHistoryJson() {
-                                    Optional<Vote> vote = decryptedBallotsWithMerits.getVote(proposal.getProposalTxId());
-                                    if (vote.isPresent())
-                                        voteJson.addProperty("vote", vote.get().isAccepted() ? "Accepted" : "Rejected");
-                                    else
-                                        voteJson.addProperty("vote", "Ignored");
-
-                                    voteJson.addProperty("voteWeight", decryptedBallotsWithMerits.getStake());
-                                    voteJson.addProperty("stake", decryptedBallotsWithMerits.getStake());
-                                    voteJson.addProperty("blindTxId", decryptedBallotsWithMerits.getBlindVoteTxId());
-                                    voteJson.addProperty("revealTxId", decryptedBallotsWithMerits.getVoteRevealTxId());
-
-                                    votesArray.add(voteJson);
-                                }));
+                        .forEach(evaluatedProposal -> {
+                            List<DecryptedBallotsWithMerits> decryptedVotesForCycle = cycle.getResultsOfCycle().getDecryptedVotesForCycle();
+                            // Make sure the votes are sorted so we can easier compare json files from different users
+                            decryptedVotesForCycle.sort(Comparator.comparing(DecryptedBallotsWithMerits::getBlindVoteTxId));
+                            decryptedVotesForCycle.forEach(decryptedBallotsWithMerits -> {
+                                JsonObject voteJson = new JsonObject();
+                                Optional<Vote> vote = decryptedBallotsWithMerits.getVote(proposal.getProposalTxId());
+                                if (vote.isPresent())
+                                    voteJson.addProperty("vote", vote.get().isAccepted() ? "Accepted" : "Rejected");
+                                else
+                                    voteJson.addProperty("vote", "Ignored");
+
+                                voteJson.addProperty("voteWeight", decryptedBallotsWithMerits.getMerit(daoStateService));
+                                voteJson.addProperty("stake", decryptedBallotsWithMerits.getStake());
+                                voteJson.addProperty("blindTxId", decryptedBallotsWithMerits.getBlindVoteTxId());
+                                voteJson.addProperty("revealTxId", decryptedBallotsWithMerits.getVoteRevealTxId());
+
+                                votesArray.add(voteJson);
+                            });
+                        });
+
+                proposalJson.addProperty("numberOfVotes", votesArray.size());
                 proposalJson.add("votes", votesArray);
-                proposalsArray.add(proposalJson);
 
+                proposalsArray.add(proposalJson);
             });
+            cycleJson.addProperty("numberOfProposals", proposalsArray.size());
```

Comment:
This seems to be a duplicate of `proposalsCount`. Please remove one of them.
If you keep this one, I suggest to also change the name of `votesCount` to `numberOfVotes` and make it an int to keep consistency.

Reply:
Ah thanks! Fixed. Can you ass ACK or utACk if it si now ok from your side